sklearn/manifold/mds.py

```diff
@@ -67,7 +67,15 @@ def _smacof_single(dissimilarities, metric=True, n_components=2, init=None,
     n_iter : int
         The number of iterations corresponding to the best stress.
     """
-    dissimilarities = check_symmetric(dissimilarities, raise_exception=True)
+    if dissimilarities.dtype == np.float16:
```

Comment:
Why these values? Why not dtype.eps? And input validation should probably not be this far down the stack.

Reply:
@Celelibi suggested using this values, so I tried using these values.

Reply:
I think using the build-in eps or some factor of that would probably be more principled?

Reply:
Okay! I will use eps rather than hard coding these values.